### PR TITLE
[Feature] Authenticated API datasource creation

### LINKED ISSUE #5782: [Feature] Add Auth API Datasource Creation option
## Summary
In new nav, give user an option to create OAuth datasource from `create new` tab.

## Motivation
It's a frequently observed use case for a lot of users.

## Additional Context
User will be able to create such datasources with a single click.

### PR DESCRIPTION
**Pull request by pranavkanade**
## Description
Added a new card to datasources pane to create REST API datasource, directly from `create new` tab.

Fixes #5782 

## Type of change
- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?
- Added a cypress test

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: feature/oauth-datasource-creation 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 51.96 **(-0.02)** | 33.55 **(-0.02)** | 30.02 **(-0.01)** | 52.62 **(-0.02)**
 :red_circle: | app/client/src/pages/Editor/DataSourceEditor/RestAPIDatasourceForm.tsx | 25.13 **(-0.64)** | 18.18 **(-1.46)** | 2.94 **(0)** | 28.14 **(-0.85)**
 :red_circle: | app/client/src/pages/Editor/DataSourceEditor/index.tsx | 27.55 **(-0.29)** | 0 **(0)** | 11.11 **(0)** | 28.74 **(-0.33)**
 :red_circle: | app/client/src/pages/Editor/IntegrationEditor/NewApi.tsx | 46.51 **(-15.99)** | 50 **(-16.67)** | 0 **(0)** | 51.28 **(-13.24)**</details>

### CODE REVIEW

**Comment by Nikhil-Nandagopal** [comment]
@pranavkanade small change, can we call this authenticated API instead of OAuth

**Comment by pranavkanade** [comment]
> @pranavkanade small change, can we call this authenticated API instead of OAuth
@Nikhil-Nandagopal you mean to choose `API Key` option for `authentication type`, right?
Screenshot 2021-07-21 at 11 42 21 AM

**Review by rishabhsaxena — DISMISSED** [review]

**Review by rishabhsaxena — APPROVED** [review]
Should we redirect to the view mode of the datasource after saving it, which has a create new query button? We could keep them on the same page as well since there's a new api button here.

Also we should keep the new api button disabled button until we save the datasource, as it still errors out even if the url is filled

**Comment by Nikhil-Nandagopal** [comment]
> Should we redirect to the view mode of the datasource after saving it, which has a create new query button? We could keep them on the same page as well since there's a new api button here.
> 
> Also we should keep the new api button disabled button until we save the datasource, as it still errors out even if the url is filled

@rishabhsaxena the new API button from edit mode needs to be removed, it's confusing. Redirecting to the view mode is to give reinforcement that it got saved and the next step is to create a new API

**Comment by pranavkanade** [comment]
@Nikhil-Nandagopal will add saved state for this.

**Review by Rishabh-Rathod — COMMENTED** [review]
1. When the user clicks on save for Authenticated API currently it stays on the Datasource Editor page itself which is not the case with most of the data sources. 

Here user needs to manually goBack to the previous page, Is it intentional?

2. When clicking on "create new" 
Screenshot 2021-07-21 at 5 47 32 PM

This newly created DataSource using Authenticated API option is visible but not selectable,
Is this expected?

**Comment by pranavkanade** [comment]
@Rishabh-Rathod this is a bug. I can see that on release as well.

**Review by rishabhsaxena — APPROVED** [review]

File: app/client/src/pages/Editor/IntegrationEditor/NewApi.tsx [hunk 1]
```diff
@@ -154,10 +154,14 @@ function NewApiScreen(props: Props) {
   }, [plugins]);
 
   const handleCreateAuthApiDatasource = useCallback(() => {
-    authApiPlugin &&
+    if (authApiPlugin) {
+      AnalyticsUtil.logEvent("CREATE_DATA_SOURCE_AUTH_API_CLICK", {
```
Comment (rishabhsaxena) [inline app/client/src/pages/Editor/IntegrationEditor/NewApi.tsx:158]: `_CLICK` I guess this suffix would be useful to differentiate b/w frontend / backend events as well

**Review by rishabhsaxena — APPROVED** [review]

**Pull request merged**